src/go/rpk/pkg/tuners/executors/commands/create_sized_file.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// +build linux
+
+package commands
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+type writeSizedFileCommand struct {
+	path      string
+	sizeBytes int64
+}
+
+func NewWriteSizedFileCmd(path string, sizeBytes int64) Command {
+	return &writeSizedFileCommand{path, sizeBytes}
+}
+
+func (c *writeSizedFileCommand) Execute() error {
+	log.Debugf("Creating  '%s' (%d B)", c.path, c.sizeBytes)
+
+	_, err := os.Stat(c.path)
+	if err == nil {
+		// If the file exists, error out to prevent overwriting important files
+		// or filesystems.
+		return errBallastExists(c.path)
+	}
+	if !os.IsNotExist(err) {
+		return err
+	}
+
+	// the 'os' package needs to be used instead of 'afero', because the file
+	// handles returned by afero don't have a way to get their file descriptor
+	// (i.e. an Fd() method):
+	// https://github.com/spf13/afero/issues/234
+	//
+	// os.O_EXCL fails if O_CREATE and the file already exists, which is an
+	// additional safety net to prevent mistakes like overwriting an
+	// important file or even a whole filesystem.
+	f, err := os.OpenFile(c.path, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0666)
```

Comment:
why not open| write | truncate? 

who cares if it was there before, if it's not the same size? 

also, it should adjust at runtime if the sizes are different.

Reply:
Cockroach does the same, actually, and I think it makes sense. From [their docs](https://www.cockroachlabs.com/docs/stable/cockroach-debug-ballast.html):
> [cockroach debug ballast] refuses to overwrite the target ballast file if it already exists. This change is intended to prevent mistaken uses of the ballast command. Consider adding an rm command to scripts that integrate cockroach debug ballast, or provide a new file name every time and then remove the old file.

It seems enough people nuked their filesystems to make that behavior necessary.

Reply:
we can give it a unique extension like `ballast.redpanda.bin` or something unique no? 

@rkruze i don't get the logical reasoning from the docs, any insight you can add. I don't follow the logic in that comment to be honest. Seems like explaining a scenario without quite saying what it is.

Reply:
@senior7515 

> It seems enough people nuked their filesystems to make that behavior necessary.

https://github.com/cockroachdb/cockroach/blob/bcc600200d848babe88c1c8163b730d1ded1c7bc/pkg/cli/debug.go#L386-L388

Reply:
The TL;DR is in the comments I wrote, too: https://github.com/vectorizedio/redpanda/pull/1818/files/e5023075703e2cca9d5c6474fe2dfc3db442bf5d#diff-deb926044a3d81d4b315df152cd94d7432a7a6e91452cce1aa4c3942876d4fb5R50-R52

Reply:
I saw you wrote that but it doesn’t answer the question

Reply:
From what I understand, the cockroach DB dev/ support team saw enough users completely destroy their filesystems by creating a ballast file at the same path of the filesystem, so they decided to guard against that by exiting if the filepath already exists.

Reply:
right, i get that is what the comment says. But my question was, if we set our own file with our own extension. `redpanda.ballast` - is not a filesystem path, why wouldn't you detect in RPK that it isn't a device filehandle, or add more guards against that. Specially in the kubernetes world, these things will be automated. So couldn't we open the file and test that it is indeed a regular file, and only issue a truncate  if it's a regular handle *and* matches the `redpanda.ballast` filename and is under the data directory? 

that comment is probably from some old manual deployments before the k8s operator or smth.

Reply:
@senior7515  I think that's a valid point, but I don't think it's too large of a burden to avoid creating a ballast if it already exists. This is a restriction we can add now, and if user experience reports are that this restriction is not too great, we can relax it and follow the scheme you're describing. We can't really do the reverse.

Although I think sooner than later we want `rpk debug ballast [ create | remove ]` to make removing a ballast a lot easier.

Reply:
I hadn't seen or heard of anyone writing over their file system with the ballast command. I could see this happening, but I agree with Alex that there are other guide rails we can put up to make this safe.

Reply:
Fixed.